### PR TITLE
Replace Debian stretch with bullseye in Helix queues

### LINKED ISSUE #57837: Remove debian 9 and add debian 11 to helix queues
Based on our official support matrix we don't support Debian 9 for .NET 6 anymore: https://github.com/dotnet/core/blob/main/release-notes/6.0/supported-os.md.

Also as Debian 11 has released, we should add it to our helix queues: 
- https://github.com/dotnet/runtime/blob/main/eng/pipelines/libraries/helix-queues-setup.yml
- https://github.com/dotnet/runtime/blob/main/eng/pipelines/coreclr/templates/helix-queues-setup.yml

Target: We need this in both release/6.0-rc1 and main. I should have opened the issue sooner...

cc @am11 in case you want to pick this up. Hopefully this is easier than the alpine 3.14 change.

### PR DESCRIPTION
**Pull request by am11**
Fixes #57837.

### CODE REVIEW

File: eng/pipelines/coreclr/templates/helix-queues-setup.yml [hunk 1]
```diff
@@ -47,17 +47,17 @@ jobs:
       - ${{ if eq(variables['System.TeamProject'], 'public') }}:
         - (Ubuntu.1804.Arm32.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm32v7-bfcd90a-20200121150440
       - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
-        - (Debian.9.Arm32)Ubuntu.1804.Armarch@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-9-helix-arm32v7-bfcd90a-20200121150037
+        - (Debian.11.Arm32)Ubuntu.1804.Armarch@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-arm32v7-20210304164347-5a7c380
```
Comment (ViktorHofer) [inline eng/pipelines/coreclr/templates/helix-queues-setup.yml:50]: Shouldn't we upgrade Debian 9 to Debian 10 as that's the min supported version?

Comment (am11) [reply eng/pipelines/coreclr/templates/helix-queues-setup.yml:50]: In libraries we have Debian 9 and Debian 10, and coreclr it's just Debian 9. I have only replaced 9. If we don't want multiple Debian queues for coreclr as well, I can change it to 10 as min version?

Comment (ViktorHofer) [reply eng/pipelines/coreclr/templates/helix-queues-setup.yml:50]: Let's have both Debian 10 and 11 everywhere (in accordance to our support matrix). If that consumes to much resources we can comment them out at any point.

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 2]
```diff
@@ -83,7 +85,7 @@ jobs:
             - (Fedora.34.Amd64.Open)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-34-helix-20210728124700-4f64125
       - ${{ if or(eq(parameters.jobParameters.interpreter, 'true'), eq(parameters.jobParameters.isSingleFile, true)) }}:
         # Limiting interp runs as we don't need as much coverage.
-        - Debian.9.Amd64.Open
+        - Debian.10.Amd64.Open
```
Comment (am11) [inline eng/pipelines/libraries/helix-queues-setup.yml:88]: > .packages/microsoft.dotnet.helix.sdk/6.0.0-beta.21419.2/tools/Microsoft.DotNet.Helix.Sdk.MonoQueue.targets(55,5): error : (NETCORE_ENGINEERING_TELEMETRY=Helix) ArgumentException: Unknown QueueId. Check that authentication is used and user is in correct groups.

@ViktorHofer, singlefile test leg is failing like this ☝️ does it mean `Debian.10.Amd64` queue does not exist?

Comment (ViktorHofer) [reply eng/pipelines/libraries/helix-queues-setup.yml:88]: Right, according to https://helix.dot.net/, there are no debian 10 or debian 11 queues. I assume we would need to use docker images. Opened an internal ticket for clarification: https://github.com/dotnet/core-eng/issues/14163.

Comment (am11) [reply eng/pipelines/libraries/helix-queues-setup.yml]: Image reference worked 👍

**Comment by ViktorHofer** [comment]
Failure is https://github.com/dotnet/runtime/issues/57196

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
@am11 you are awesome 🥇

**Pull request merged**